src/macros.rs

```diff
@@ -89,6 +89,17 @@ macro_rules! algos {
                     ),
                 }
             }
+
+            /// Creates a new encoder, using the specified compression level and parameters, which
+            /// will read uncompressed data from the given stream and emit a compressed stream.
+            pub fn with_quality_and_params(inner: $inner, level: crate::Level, params: &[libzstd::stream::raw::CParameter]) -> Self {
```

Comment:
I think it would be a good idea to re-export `libzstd::stream::raw::CParameter` with a name like `ZstdRawParameter` so that users of this crate do not have to import `zstd` again and can keep in sync with the zstd version used by this crate?

Reply:
Such a re-export would be the first of its kind for this crate.

Reasoning explained here: https://github.com/Nullus157/async-compression/issues/133#issuecomment-979959570

Reply:
But the API here would be the same as re-export, upgrading zstd would require our users to also upgrade their own zstd version if they use this API, which essentially mean it's the same as a breaking change.

Reply:
I'm also ok with a newtype here, though it would require extra work.

Reply:
I'm happy to provide a newtype here, if that's the preferred approach.

I also think re-exporting the zstd type would likely be safe, as zstd isn't going to break ABI. But I'm happy to go with whichever approach you'd prefer.

Reply:
I'd prefer a newtype here, since users could try to pass a `zstd::stream::raw::CParameter` of a different `zstd` version than the one used in our crate and that will cause a hard compile-time error.